modules/loadbalancer/variables.tf

```diff
@@ -19,27 +19,43 @@ variable "tags" {
   type        = map(string)
 }
 
-variable "zones" {
+variable "load_balancer" {
```

Comment:
To be honest I'm still not convinced about grouping all the rest of variables 😃 I mean it makes sense to me to group directly related variables like in the `image` or `virtual_machine` variables in the "vmseries" module but it doesn't look good to me when we have a nested common group with the same name as the module as in "application_gateway" or "load_balancer" modules.

I would say we group variables according to the Azure Portal interface along with only directly related variables for the rest. Like the `network` or `bgp` variables in the VNG module, or the `virtual_machine` variable in "vmseries". And we can leave the rest as separate variables as before. Otherwise if we go with grouping all the variables approach I don't see a clear way of what to keep out of this scope (like in the subnet_id case)

Btw there was also the case with Terraform Cloud where complex type variables need to be written on a large text area which wasn't convenient, I'm not sure if that's still the case but it might be something important to consider as well.

Reply:
I partially agree with you hence I was thinking to eliminate this `load_balancer` variable in LB module. But at the same time, all other map variables in this module like `frontend_ips` and `health_probes` make perfect sense to me, would you agree?

In AppGW module it's a little bit more complicated but I think we could try to split `application_gateway` into some smaller maps (like 'global_ssl_policy') and some single variables as well.

Reply:
Yes it also makes sense to me to keep variables grouped like `frontend_ips` - my only argument is to avoid grouping all the rest of the variables as a sub-group with the same name as the module. I see no harm keeping these variables at the root level.
I also propose keeping `zones` variable at the root level for all the modules. As far as I can see, we always apply the same "zone" for the resources in a module (if it allows).

Reply:
@alperenkose & @sebastianczech - I refactored variables for _appgw_, _lb_, _natgw_ & _vng_ again. Please take a look and see whether it makes sense to you now.